executor/runtime/docker/docker.go

```diff
@@ -1955,32 +1988,32 @@ func (r *DockerRuntime) Kill(c *runtimeTypes.Container) error { // nolint: gocyc
 		goto stopped
 	}
 
-	if err := r.client.ContainerStop(context.TODO(), c.ID, &containerStopTimeout); err != nil {
+	if err := r.client.ContainerStop(context.TODO(), r.c.ID, &containerStopTimeout); err != nil {
```

Comment:
Same as above

Reply:
No, we can't, because by the time that this part (Kill) gets called, the parent context may already have been cancelled (hence why we're killing. I don't yet know how to solve this problem. Suggestions welcome.